Use after free as the copied prompt is allocated in check_user_interactive;
this later causes a crash when we ask for a session annotation.
Reset it to the standard password prompt.

Reported to upstream: https://bugzilla.sudo.ws/show_bug.cgi?id=967

diff -r sudo-1.8.31p1/plugins/sudoers/auth/pam.c sudo-1.8.31p1-p/plugins/sudoers/auth/pam.c
--- sudo-1.8.31p1/plugins/sudoers/auth/pam.c	2019-10-28 12:27:38.000000000 +0000
+++ sudo-1.8.31p1/plugins/sudoers/auth/pam.c	2021-03-03 11:49:47.405428118 +0000
@@ -300,6 +300,8 @@
 
     /* PAM_SILENT prevents the authentication service from generating output. */
     *pam_status = pam_authenticate(pamh, PAM_SILENT);
+    /* reset def_prompt as prompt may have been allocated and later freed. */
+    def_prompt = PASSPROMPT;
     if (getpass_error) {
 	/* error or ^C from tgetpass() */
 	debug_return_int(AUTH_INTR);
